### ISSUE TITLE
[suggestion] Monkey friendly examples 

### ISSUE THREAD

**Issue by Izopi4a** [dgrijalva/jwt-go#206]
Hello, first thanks for the wonderful package.

Because I suck, I spend ~ 30mins figuring in this example of checking the token

```
token, err := jwt.Parse(t.Token, func(token *jwt.Token) (interface{}, error) {
        return []byte("AllYourBase"), nil
    })
```

that []byte("AllYourBase") should be my secret. 

It is TOTALLY my fault, i suck, but if others agree you could change that with "token_key_goes_here" 

Cheers

**Comment by tobert** [comment]
Agreed. For my code I opted to use a regular function and it's more clear.

```go
// signingKeyCb is a callback for jwt.Parse. The examples show it as
// an anonymous function, but the "return secret, nil" freaked me out at
// a glance, so here it is as a full function to make things clear.
//
// The examples & docs show a type check for the signature, which is
// probably useful if multiple signing methods are in play, but this
// code only supports one, so I've decided against including it.
func signingKeyCb(t *jwt.Token) (interface{}, error) {
    secret, err := getSigningSecret()
    if err != nil {
        return "", err // TODO: annotate
    }
    
    return secret, nil
}

func parseJWT(tokenStr string) (out MyTokenThing, error) {
    out.jwt, err = jwt.ParseWithClaims(tokenStr, jwt.StandardClaims{}, signingKeyCb)
    ...
}
```

**Comment by dgrijalva** [comment]
I've never been very good at writing examples.  Reading the godocs is far more useful than following the examples.  If you want to open a PR with clearer examples, I'd be happy to review them.

**Comment by dgrijalva** [comment]
I've always felt this design was a bit clunky. I've seen other libraries where you just pass in the key, but there are a lot of use cases where you need information in the header in order to provide the correct key.

I've seen implementations where different signing methods are used for different type of client/operation. There's also the [`kid`](https://tools.ietf.org/html/rfc7515#section-4.1.4) field in the spec (among others).

@tobert I also tend to use regular functions for this, though I've seen cases where the closure style is helpful. Thank you for the useful example.

I'm going to include this ticket in v4 as a reminder to clean up some of these examples.
